File: Fondamenti_programmazione_II/Java/EserciziEsame/GestioneCircoloSportivo/Socio.java
package EserciziEsame.GestioneCircoloSportivo;

import java.util.*;

public class Socio {
    
    private int codice, eta, categoria;
    private String nomeCognome;
    protected List<Prenotazione> prenotazioni;

    public Socio (int codice, String nomeCognome, int eta, int categoria) {
        this.codice = codice;
        this.nomeCognome = nomeCognome;
        this.eta = eta;
        this.categoria = categoria;
        prenotazioni = new LinkedList<Prenotazione>();
    }

    public void addPrenotazione (Prenotazione p) {
        prenotazioni.add(p);
    }

    public String toString () {
        return codice + "\t" + nomeCognome + "\t" + 
        eta + "\t" + categoria + "\t" + prenotazioni + "\t";
    }
}

File: Fondamenti_programmazione_II/Python/Esami/Gestione_circolo_sportivo/Socio.py
class Socio:
    def __init__(self, codice, nome, cognome, eta, categoria):
        self._codice = codice
        self._cognome = cognome
        self._nome = nome
        self._eta = eta
        self.categoria = categoria
        self._prenotazioni = []

    def addPrenotazione(self, prenotazione):
        self._prenotazioni.append(prenotazione)
    
    def getPrenotazioni(self):
        return self._prenotazioni

    def __str__(self):
        return str(self._codice) +"\t"+ self._nome +"\t"+ self._cognome +"\t"+ str(self._eta) +"\t"+ str(self.categoria) +"\t"+ str([str(p) for p in self._prenotazioni])